### ISSUE TITLE
Position absolute inside relative container does not work properly?

### ISSUE THREAD

**Issue by kamalsamra** [dompdf/dompdf#1083]
Position absolute take body as relative container it does not take my custom relative container as relative element.
my HTML structure is 
<html>
<head>
<style>
.first {
  background-color: #ddd;
  padding: 10px;
}
.second {
  position: relative;
  background-color: #f00;
  height: 500px;
}
.absolute1 {
  position: absolute;
  width:300px;
  top: 0;
  bottom: 0;
  background-color: #ff0;
}
.absolute2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 300px;
  right: 200px;
  background-color: #3c3c3c;
}
.third {
  background-color: #555;
  padding: 10px;
}
</style>
</head>
<body>
<div class="wrapper">
  <div class="first">
    <p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p>
  </div>
  <div class="second">
    <div class="absolute1"></div>
    <div class="absolute2"></div>
  </div>
  <div class="third">
    <p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p><p>kjsafddsajkfghsdkjgbhsdkfghsdjkfgh</p>
  </div>
</div>
</body>
</html>
you can see its output here 
http://eclecticgeek.com/dompdf/debug.php?identifier=749ee80d12d4da379e3f0faa91674961

**Comment by bsweeney** [comment]
Yes, it looks like the calculate for `bottom: 0;` is off. If you're specifying the height of the containing element then to work around the issue you can just specify the height instead.

**Comment by mrtuvn** [comment]
@bsweeney Hi. How to setup sepecific div alway bottom of last page.
I have try to use css code like
#wholepage {position:relative}
#wholepage .footer {position:absolute; height:xxpx; bottom:0} //show on every bottom page
But seem it not works. If div contain table it will lost all border

**Comment by bsweeney** [comment]
@mrtuvn can you take your question to the [support group](http://groups.google.com/group/dompdf)? I'm not entirely sure what you need to fix. Some sample HTML would also be useful.

**Comment by arifhussain353** [comment]
@bsweeney  Hi,  
I don't know if he is talking about the same problem which i am facing also. Below is little bit explanation of related: 
html 
```
<div class="sheet" style="position:relative">
   <div class="item"..... 
   <div class="item"..... 
  <div class="item"..... 
</div>
```
Php : 
```
  public function getPDF(Request $request){
        $html = '<div class="item col_3_block ui-draggable ui-draggable-handle ui-draggable-dragging 
                      new_item" style="position: absolute; left: 470px; top: 260px;"><div 
                      class="name">col_3_block1</div></div>';

        $pdf = App::make('dompdf.wrapper');
        $pdf->loadHTML($html)->setPaper('a4', 'portrait');

        return $pdf->stream();
    }
```
[**Screenshot 1**.](https://prnt.sc/gn33a0)
In html css  : screenshot 1 i simply drag and drop element from left to right A4 side white box which have height and width equal to standard A4 size page amd `relative` position.
css style 
`position: absolute; left: 470px; top: 260px;`

[**Screenshot 2**](https://prnt.sc/gn3jq4)
After pdf generate the position of same element is little bit disturbed.

**Problem**
Screenshot 2  position in `PDF`  is not accurate  according to  screenshot 2 position in html css.

**Comment by AllanWegan** [comment]
Also had the problem of absolute positioning inside a relatively positioned element using the page as reference for offsets from bottom.
I fixed it for me by setting $needs_reposition to false right after its only assignment in dompdf/src/FrameReflower/Block.php.

**Comment by bsweeney** [comment]
@AllanWegan thanks for the pointer. We'll see if this is implicated.
